File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
import heapq
def solution(scoville, K):
    heap = []
    for i in range(len(scoville)):
        heapq.heappush(heap, scoville[i])
    
    answer = 0
    while heap[0] < K:
        if len(heap) == 1:
            answer = -1
            break
        a = heapq.heappop(heap)
        b = heapq.heappop(heap)
        heapq.heappush(heap, a+b*2)
        answer += 1
        
    return answer

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.io.*;
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        PriorityQueue<Integer> hq = new PriorityQueue<>();
        for(int i : scoville){
            hq.add(i);
        }
        while(hq.peek() != null){
            if(hq.peek() >= K){
                break;
            }
            int first = hq.poll();
            if(hq.peek() == null){
                answer = -1;
                break;
            }
            int second = hq.poll();
            hq.add(first+second*2);
            answer ++;
        }
        
        return answer;
    }
}